### PR TITLE
Feature/reader stream refresh cards photo

### PR DESCRIPTION
**Pull request by nbradbury**
Adds support for "photo cards" as part of the reader stream refresh (#4776). Posts that have a featured image and less than 100 characters of text now show their title on the featured image itself and no other content is shown.

An example is the first post in this screenshot:

screenshot_1479498423

Note: An easy way to test this is to follow a tag that's likely to have a lot of photos, such as "sunrise" or "nature."

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/models/ReaderPost.java [hunk 1]
```diff
@@ -207,6 +208,10 @@ public static ReaderPost fromJson(JSONObject json) {
             post.setRailcarJson(jsonRailcar.toString());
         }
 
+        // set the card type last since it depends on information contained in the post - note
+        // that this is stored in the post table rather than calculated on-the-fly
```
Comment (oguzkocer) [inline WordPress/src/main/java/org/wordpress/android/models/ReaderPost.java]: Thanks a lot for this comment. Quick question, is the `fromJson` handled in a background thread? If not, running this multiple times might be a costly operation especially due to stripping of text for the calculation.

File: WordPress/src/main/java/org/wordpress/android/models/ReaderCardType.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    /*
+     * returns true if the post's content is 100 characters or less
+     */
+    private static boolean hasMinContent(@NonNull ReaderPost post) {
+        if (post.getExcerpt().length() > MIN_CONTENT_CHARS) {
+            return false;
+        }
+        if (post.getText().length() <= MIN_CONTENT_CHARS) {
```
Comment (oguzkocer) [inline WordPress/src/main/java/org/wordpress/android/models/ReaderCardType.java]: Right, right. Sorry. I completely misunderstood this method. Please disregard the following comment as well.

File: WordPress/src/main/res/layout/reader_cardview_post.xml [hunk 3]
```diff
@@ -81,10 +81,35 @@
...
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_gravity="bottom|left"
+                android:layout_marginBottom="@dimen/margin_extra_large"
+                android:layout_marginLeft="@dimen/reader_card_content_padding"
+                android:layout_marginRight="@dimen/reader_card_content_padding"
+                android:ellipsize="end"
+                android:maxLines="1"
+                android:shadowColor="@color/grey_dark"
+                android:shadowDx="3.0"
```
Comment (oguzkocer) [inline WordPress/src/main/res/layout/reader_cardview_post.xml]: It's minor, but we could consider extracting these (`shadowDx`, `shadowDy`, `shadowRadius`) into `dimens.xml`.

File: WordPress/src/main/java/org/wordpress/android/models/ReaderPost.java (hunk 1 above)
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/models/ReaderPost.java]: Yes, posts are parsed in the background - see [here](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/reader/services/ReaderPostService.java#L278), for example.

File: WordPress/src/main/java/org/wordpress/android/models/ReaderCardType.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            return true;
+        }
+        return (HtmlUtils.fastStripHtml(post.getText()).length() <= MIN_CONTENT_CHARS);
+    }
+
+    public static String toString(ReaderCardType cardType) {
+        if (cardType == null) {
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/models/ReaderCardType.java]: I'm working on a separate PR which adds `VIDEO` card type :)

File: WordPress/src/main/res/layout/reader_cardview_post.xml (hunk 3 above)
Comment (nbradbury) [inline WordPress/src/main/res/layout/reader_cardview_post.xml]: Added in 15b1c42

**Comment by nbradbury** [comment]
Thanks for the review @oguzkocer - ready for round two.

**Comment by oguzkocer** [comment]
Thanks @nbradbury, looks good :shipit:

**Pull request merged**